### PR TITLE
[Serve] [Docs] Add Serve REST API Schema to Serve API Docs

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

#25691 updates the Serve CLI and REST API. This change adds the new REST API schema to the documentation, including info about endpoints, REST requests, and the input and output JSON schemas.

Link to this change's documentation: https://ray--25786.org.readthedocs.build/en/25786/serve/package-ref.html#serve-rest-api

## Related issue number

N/A

## Checks

- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - N/A: this is a documentation change.

### CODE REVIEW

**Comment by simon-mo** [comment]
re: httpdomain
image

if we are _just_ rendering these with the new package, it seems a little bit excessive? I was under the impression there are more ux features as well. But looks like just simple headings? 


re: CLI, great job!

File: doc/source/serve/package-ref.md [hunk 1]
```diff
@@ -45,3 +45,63 @@
...
+
+```
+
+#### REST API
+
+```{eval-rst}
+.. http:GET:: "/api/serve/deployments/"
+    :noindex:
+
+        Gets latest config that Serve has received. This config represents the current goal state for the Serve application. Starts a Serve application on the Ray cluster if it's not already running. See Config Schema for the output schema.
```
Comment (edoakes) [inline doc/source/serve/package-ref.md:66]: link to "config schema" and don't capitalize. also goes for other instances

File: doc/source/serve/package-ref.md [hunk 2]
```diff
@@ -45,3 +45,63 @@
...
+```
+
+```{eval-rst}
+.. http:PUT:: "/api/serve/deployments/"
+    :noindex:
+
+        Uploads config for Serve application to deploy. Starts a Serve application on the Ray cluster if it's not already running. See Config Schema for the request schema.
```
Comment (edoakes) [inline doc/source/serve/package-ref.md:73]:
```suggestion
        Declaratively deploys the Serve application. Starts Serve on the Ray cluster if it's not already running. See Config Schema for the request schema.
```

**Review by edoakes — COMMENTED** [review]
just nits

File: doc/source/serve/package-ref.md (hunk 1 above)
Comment (shrekris-anyscale) [reply doc/source/serve/package-ref.md]: Good catch– I made the updates.

**Comment by shrekris-anyscale** [comment]
> if we are just rendering these with the new package, it seems a little bit excessive? I was under the impression there are more ux features as well. But looks like just simple headings?

@simon-mo Good points– I removed the dependency and replaced the formatting with Markdown. I've also added some example requests and responses to all the REST API methods.

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
@shrekris-anyscale lint failure

**Review by simon-mo — APPROVED** [review]

**Pull request merged**